[package] libjxl/*: usage in libvips fails with link error

### Description

When building libvips with the option `with_jpeg_xl=True` the linking of the libvips executables fails, due to a unresolved external symbol `JxlGetDefaultCms`

I was first thinking this is a libvips issue, [but it seems like that is not the case](https://github.com/libvips/libvips/issues/4351)

So currently my guess is that this is a libjxl issue?

I tested using `JxlGetDefaultCms()` in the test package for libjxl, which worked. (Maybe the problem is also that libvips uses meson, while the test package uses cmake?)

I tested versions `0.11.1` and `0.6.1` but I think this might be more of a conan recipe issue?

### Package and Environment Details

* Package Name/Version: **libjxl/0.11.1**
* Operating System+version: **Windows 11**
* Compiler+version: **MSVC 19.42.34433.0**
* Conan version: **conan 2.11.0**
* Python version: **Python 3.11.3**


### Conan profile

[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.version=194
os=Windows



### Steps to reproduce

`conan create all/conanfile.py --version=8.16.0 -o "libvips/*:with_jpeg_xl=True"`

### Logs

<details><summary>Click to expand log</summary>

```
[380/383] Linking target tools/vipsedit.exe
FAILED: tools/vipsedit.exe
"link" @tools/vipsedit.exe.rsp
   Creating library tools\vipsedit.lib and object tools\vipsedit.exp
jxl.lib(encode.obj) : error LNK2019: unresolved external symbol JxlGetDefaultCms referenced in function JxlEncoderCreate
jxl.lib(enc_jpeg_data.obj) : error LNK2001: unresolved external symbol JxlGetDefaultCms
tools\vipsedit.exe : fatal error LNK1120: 1 unresolved externals
[381/383] Linking target tools/vips.exe
FAILED: tools/vips.exe
"link" @tools/vips.exe.rsp
   Creating library tools\vips.lib and object tools\vips.exp
jxl.lib(encode.obj) : error LNK2019: unresolved external symbol JxlGetDefaultCms referenced in function JxlEncoderCreate
jxl.lib(enc_jpeg_data.obj) : error LNK2001: unresolved external symbol JxlGetDefaultCms
tools\vips.exe : fatal error LNK1120: 1 unresolved externals
[382/383] Linking target tools/vipsthumbnail.exe
FAILED: tools/vipsthumbnail.exe
"link" @tools/vipsthumbnail.exe.rsp
   Creating library tools\vipsthumbnail.lib and object tools\vipsthumbnail.exp
jxl.lib(encode.obj) : error LNK2019: unresolved external symbol JxlGetDefaultCms referenced in function JxlEncoderCreate
jxl.lib(enc_jpeg_data.obj) : error LNK2001: unresolved external symbol JxlGetDefaultCms
tools\vipsthumbnail.exe : fatal error LNK1120: 1 unresolved externals
[383/383] Linking target tools/vipsheader.exe
FAILED: tools/vipsheader.exe
"link" @tools/vipsheader.exe.rsp
   Creating library tools\vipsheader.lib and object tools\vipsheader.exp
jxl.lib(encode.obj) : error LNK2019: unresolved external symbol JxlGetDefaultCms referenced in function JxlEncoderCreate
jxl.lib(enc_jpeg_data.obj) : error LNK2001: unresolved external symbol JxlGetDefaultCms
tools\vipsheader.exe : fatal error LNK1120: 1 unresolved externals
ninja: build stopped: subcommand failed.

libvips/8.16.0: ERROR: 
Package '55e22a47953bd24f607f9bdb599ace51870b7ba8' build failed
libvips/8.16.0: WARN: Build folder C:\Users\examp\.conan2\p\b\libvi77860d0ece1e5\b\build-release
ERROR: libvips/8.16.0: Error in build() method, line 322
        meson.build()
        ConanException: Error 1 while executing
```

</details>

--- a/recipes/libjxl/all/conanfile.py
+++ b/recipes/libjxl/all/conanfile.py
@@ -9,7 +9,7 @@ from conan.tools.gnu import PkgConfigDeps
 from conan.tools.microsoft import is_msvc
 from conan.tools.scm import Version
 
-required_conan_version = ">=1.53.0"
+required_conan_version = ">=2"
 
 
 class LibjxlConan(ConanFile):
@@ -199,6 +199,7 @@ class LibjxlConan(ConanFile):
                 self.cpp_info.components["jxl"].defines.append("JXL_CMS_STATIC_DEFINE")
             if libcxx:
                 self.cpp_info.components["jxl_cms"].system_libs.append(libcxx)
+            self.cpp_info.components["jxl"].requires.append("jxl_cms")
 
         # jxl_dec
         if Version(self.version) < "0.9.0":

--- a/recipes/libvips/all/conanfile.py
+++ b/recipes/libvips/all/conanfile.py
@@ -154,7 +154,7 @@ class LibvipsConan(ConanFile):
         elif self.options.with_jpeg == "mozjpeg":
             self.requires("mozjpeg/4.1.5")
         if self.options.with_jpeg_xl:
-            self.requires("libjxl/0.6.1")
+            self.requires("libjxl/0.11.1")
         if self.options.with_lcms:
             self.requires("lcms/2.16")
         if self.options.with_magick:
@@ -218,9 +218,9 @@ class LibvipsConan(ConanFile):
             raise ConanInvalidConfiguration("librsvg recipe not available in conancenter yet")
 
     def build_requirements(self):
-        self.tool_requires("meson/1.4.0")
+        self.tool_requires("meson/[>=1.2.3 <2]")
         if not self.conf.get("tools.gnu:pkg_config", check_type=str):
-            self.tool_requires("pkgconf/2.1.0")
+            self.tool_requires("pkgconf/[>=2.2 <3]")
         if self.options.introspection:
             self.tool_requires("gobject-introspection/1.72.0")
         self.tool_requires("glib/<host_version>")
@@ -303,7 +303,7 @@ class LibvipsConan(ConanFile):
 
     def _patch_sources(self):
         apply_conandata_patches(self)
-        
+
         # Disable tests
         meson_build = os.path.join(self.source_folder, "meson.build")
         replace_in_file(self, meson_build, "subdir('test')", "")

